### PR TITLE
Use SDK 3.0 for the integration tests, re-enable failing tests, remove redundant tests

### LINKED ISSUE #8155: Reenable FrameworkReference pack tests
The two tests for packing projects with FrameworkReferences (PackCommand_PackProject_PacksFrameworkReferences &
 PackCommand_PackProject_PacksFrameworkReferences_FrameworkReferencesAreCaseInsensitive) are disabled for now.  They need to be enabled in branch  dev-feature-xplatVerification when targeting to netstandar2.1.

1.Remove skip notation for testing PackCommand_PackProject_PacksFrameworkReferences &
 PackCommand_PackProject_PacksFrameworkReferences_FrameworkReferencesAreCaseInsensitive
2.Change the name of FrameworkReference if necessary (should match the name with the names in knownFrameworkReference provided by CLI team).

### LINKED ISSUE #8423: Enable test PackCommand_WithGeneratePackageOnBuildSet_CanPublish for netcoreapp3.0 when it's ready
Test PackCommand_WithGeneratePackageOnBuildSet_CanPublish is broken for SDK issue https://github.com/dotnet/sdk/issues/3367  (for netcoreapp3.0 only).
We skip this test for now.
We need to enable this test for netcoreapp3.0 when the above issue is solved.

### LINKED ISSUE #8440: Tests should run against 3.0 SDK
We insert into 3.0. We should fix up the test infrastructure so that we test against that as well. 

It will also allow us to enable. DotnetListPackage_DeprecatedAndOutdated_Fail. 

//cc @NuGet/nuget-client @xavierdecoster

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/8440
Fixes: https://github.com/NuGet/Home/issues/8423
Fixes: https://github.com/NuGet/Home/issues/8155

Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
Use 3.0 SDK for testing. Re-enable failing tests (note that this change also re-enables some tests @heng-liu  has had to disable in the xplat signing feature branch). 

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  Test infrastructure work
Validation:

### CODE REVIEW

**Comment by nkolev92** [comment]
@heng-liu 

This also overlaps with some of your work. 
Once I merge this to dev, I can rebase the feature branch off of dev and resolve all the conflicts.

**Review by zivkan — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs]: Add named parameter for the pesky literal `true`.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs]: Fix now or remove comment and open an issue.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (dtivel) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs:204]: Why is this now Windows-only?

**Review by dtivel — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs:204]: It was always supposed to be Windows only. 
The patching infrastructure is not set up on linux/mac.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (donnie-msft) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: Why is this string slightly different, and can they both be the same and used in lieu of the hard coded string in the test? 
`ListPkg_InvalidOptionsOutdatedAndDeprecated = "Invalid command. Combining the '--outdated' and '--deprecated' options is not supported"`

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
Comment (donnie-msft) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs]: Why not catch the exception type you want rather than cast it?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
Comment (donnie-msft) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs:3960]: Test removed because... it was specific to net461 and no longer needed?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: That's what got added in the SDK. 
I will create a separate issue to align these.

**Review by heng-liu — COMMENTED** [review]
If there're two folders(eg: netstandard2.0 & netstandard2.1), we only need to copy the dll under netstandard2.1, right?

**Review by heng-liu — CHANGES_REQUESTED** [review]
If there're two folders(eg: netstandard2.0 & netstandard2.1), we only need to copy the dll under netstandard2.1, right?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
Comment (heng-liu) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs:281]: If there're two folders(eg: netstandard2.0 & netstandard2.1), we only need to copy the dll under netstandard2.1, right?

**Review by heng-liu — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: https://github.com/NuGet/Home/issues/8454

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs]: It was copied from Heng's branch so I didn't want merge conflicts, but I can fix it up :)

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs]: It was copied from Heng's branch so I didn't want merge conflicts, but I can fix it up :) 
Few people provided similar feedback on the method, so worth fixing :)

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs:3960]: No longer needed. 
This was added as an integration test for the E2E scenario. 
The SDK now validates this on their end as well. 

This test failed with other errors, so it doesn't even hit the codepath we want to test anymore.

**Comment by nkolev92** [comment]
Blocked on the [Mac pool upgrade.](http://ddweb/techease/TicketEdit.aspx?id=1622950)

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (xavierdecoster) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: PR out to fix the messaging: https://github.com/NuGet/NuGet.Client/pull/3013

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
Comment (zkat) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs]: This issue seems to be fixed already. Can we remove this now?

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs
Comment (zkat) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs]: The PR has been merged. Can we go back to the non-literal now?

**Review by zkat — APPROVED** [review]

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs
Comment (nkolev92) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs]: Yup, will do.

**Comment by nkolev92** [comment]
Currently there are 4 failing tests due to machine changes. 

All of those are signing related. Considering disabling those tests.

**Comment by nkolev92** [comment]
Disabled 4 tests that relate to signing to xplat. 
Tracking issue: https://github.com/NuGet/Home/issues/8047

With that, I think this PR is ready to merge as soon as the mac machine migration is complete.

**Pull request merged**
